documentation/library_guide/kernel_templates/esimd/radix_sort.rst

```diff
@@ -0,0 +1,322 @@
+      sycl::buffer<char> values{sycl::range<1>(n)};
+
+      // initialize
+      {
+         sycl::host_accessor k_acc{keys, sycl::write_only};
+         k_acc[0] = 3, k_acc[1] = 2, k_acc[2] = 1, k_acc[3] = 5, k_acc[4] = 3, k_acc[5] = 3;
+
+         sycl::host_accessor v_acc{values, sycl::write_only};
+         v_acc[0] = 'r', v_acc[1] = 'o', v_acc[2] = 's', v_acc[3] = 'd', v_acc[4] = 't', v_acc[5] = 'e';
+      }
+
+      // sort
+      auto e = kt::esimd::radix_sort_by_key<true, 8>(q, keys, values, kt::kernel_param<96, 64>{}); // (3)
+      e.wait();
+
+      // print
+      {
+         sycl::host_accessor k_acc{keys, sycl::read_only};
+         for(std::size_t i = 0; i < n; ++i)
+               std::cout << k_acc[i] << ' ';
+         std::cout << '\n';
+
+         sycl::host_accessor v_acc{values, sycl::read_only};
+         for(std::size_t i = 0; i < n; ++i)
+               std::cout << v_acc[i] << ' ';
+         std::cout << '\n';
+      }
+
+      return 0;
+   }
+
+**Output:**
+
+.. code:: none
+
+   1 2 3 3 3 5
+   s o r t e d
+
+
+-----------------------------------------
+Recommended Settings for Best Performance
+-----------------------------------------
+
+The general advice is to set your configuration according to the performance measurements and profiling information. The initial configuration may be selected according to these points:
+
+- When the number of elements to sort is small (~16K or less) and the algorithm is ``radix_sort``, then the elements can be processed by a single work-group. Increase the param values, so ``N <= param.data_per_workitem * param.workgroup_size``.
+
+- When the number of elements to sort is medium (between ~16K and ~1M), then all the work-groups can execute simultaneously. Make sure the device is saturated: ``param.data_per_workitem * param.workgroup_size ≈ N / device_xe_core_count``. A larger ``param.workgroup_size`` in ``param.data_per_workitem * param.workgroup_size`` combination is preferred to reduce the number of work-groups and the synchronization overhead.
+
+- When the number of elements to sort is large (more than ~1M), then the work-groups preempt each other. Increase the occupancy to hide the latency with ``param.data_per_workitem * param.workgroup_size ≈< N / (device_xe_core_count * desired_occupancy)``. The occupancy depends on the local memory usage, which is determined by ``key_type``, ``value_type``, ``radix_bits``, ``param.data_per_workitem`` and ``param.workgroup_size`` parameters. Refer to :ref:`Local Memory <local-memory>` section for the calculation.
+
+
+.. _limitations:
+
+-----------
+Limitations
+-----------
+
+- Algorithms can only process C++ integral and floating-point types with a width of up to 64 bits (except for ``bool``).
+- Number of elements to sort must not exceed `2^30`.
+- ``radix_bits`` can only be `8`.
+- ``param.data_per_workitem`` has discreteness of `32`.
+- ``param.workgroup_size`` can only be `64`.
```

Comment:
Should we provide a note in the "Recommended Settings for Best Performance" section that `workgroup_size` is fixed at 64, and therefore the only "knob" users currently have access to in order to follow these guidelines is `data_per_workitem`?

Reply:
Adding a note of explanation makes sense to me.

Reply:
I think it is better to simplify this section omitting `workgroup_size` parameter. I will preserve the current description in case of adding more options for `workgroup_size`.

Reply:
I'he added a note